OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -264,10 +244,16 @@ bool IsBlockedBy(Actor self, Actor otherActor, Actor ignoreActor, CellConditions
 			if (self == null)
 				return true;
 
+			// If the check allows: We are not blocked by units that we can force to move out of the way.
+			if (check < BlockedByActor.NotMoving &&
+				self.Owner.Stances[otherActor.Owner] == Stance.Ally &&
```

Comment:
This is an extremely hot method, so these extra checks could potentially cause a significant perf regression during blob-on-blob battles - need to profile this!

Reply:
Could do some benchmarks around this, but would be nice to have #16328 first